ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -0,0 +1,3018 @@
+          checkDebugPrinting: false)
+        return
+      }
+      let kbID = keyboard[Key.keyboardId]!
+      let kbVersion = keyboard[Key.keyboardVersion]
+      let isUpdate = (request.userInfo[Key.update] as? Int)! != 0
+
+      if let statusCode = request.responseStatusCode, statusCode == 200 {
+        // The request has succeeded.
+        if downloadQueue!.requestsCount == 0 {
+          // Download queue finished.
+          downloadQueue = nil
+          registerCustomFonts()
+          kmLog("Downloaded keyboard: \(kbID).", checkDebugPrinting: true)
+
+          NotificationCenter.default.post(name: .keymanKeyboardDownloadCompleted, object: self,
+                                          userInfo: request.userInfo)
+          if isUpdate {
+            shouldReloadKeyboard = true
+            reloadKeyboard()
+          }
+          let userData = activeUserDefaults()
+          userData.set([Date()], forKey: Key.synchronizeSWKeyboard)
+          userData.synchronize()
+        }
+      } else { // Possible request error (400 Bad Request, 404 Not Found, etc.)
+        downloadQueue!.cancelAllOperations()
+        downloadQueue = nil
+
+        let errorMessage = "\(request.responseStatusMessage ?? ""): \(request.url)"
+        let error = NSError(domain: "Keyman", code: 0,
+                            userInfo: [NSLocalizedDescriptionKey: errorMessage])
+        kmLog("Keyboard download failed: \(error).", checkDebugPrinting: true)
+
+        if !isUpdate {
+          let fileName = request.url.lastPathComponent
+          if fileName.hasJavaScriptExtension {
+            if let kbPath = keyboardPath(forFilename: fileName, keyboardVersion: kbVersion) {
+              try? FileManager.default.removeItem(at: kbPath)
+            }
+          } else if fileName.hasFontExtension {
+            // TODO: Why do we delete a keyboard with that name?
+            if let kbPath = keyboardPath(forFilename: fileName, keyboardVersion: kbVersion) {
+              try? FileManager.default.removeItem(at: kbPath)
+            }
+            if let fontPath = activeFontDirectory()?.appendingPathComponent(fileName) {
+              try? FileManager.default.removeItem(at: fontPath)
+            }
+          }
+        }
+        downloadFailed(forKeyboard: request.userInfo, error: error)
+      }
+    case .downloadCachedData:
+      if request == currentRequest {
+        do {
+          let responseDict = try JSONSerialization.jsonObject(with: request.rawResponseData!,
+                                                              options: .mutableContainers) as? [AnyHashable: Any]
+          options = responseDict?[Key.options] as? [AnyHashable: Any] ?? [:]
+          let unsortedLanguages = (responseDict?[Key.languages] as? [AnyHashable: Any])?[Key.languages]
+            as? [[String: Any]]
+          languages = unsortedLanguages!.sorted { a, b -> Bool in
```

Comment:
It seems odd that we're allowing nil values before this point in the branch but suddenly expect to have a definite instance on this line.

Reply:
Changing to force the value in the previous line.